app/version.go

```diff
@@ -0,0 +1,94 @@
+package app
+
+import (
+	"fmt"
+
+	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
+	v2 "github.com/celestiaorg/celestia-app/pkg/appconsts/v2"
+	"github.com/celestiaorg/celestia-app/x/blob"
+	"github.com/celestiaorg/celestia-app/x/mint"
+	"github.com/celestiaorg/celestia-app/x/qgb"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	authzmodule "github.com/cosmos/cosmos-sdk/x/authz/module"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/capability"
+	"github.com/cosmos/cosmos-sdk/x/crisis"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/evidence"
+	feegrantmodule "github.com/cosmos/cosmos-sdk/x/feegrant/module"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/ibc-go/v6/modules/apps/transfer"
+	ibc "github.com/cosmos/ibc-go/v6/modules/core"
+)
+
+var (
+	// versions that the current state machine supports
+	supportedVersions = []uint64{v1.Version, v2.Version}
+
+	v1moduleVersionMap = module.VersionMap{
+		"bank":         bank.AppModule{}.ConsensusVersion(),
+		"auth":         auth.AppModule{}.ConsensusVersion(),
+		"authz":        authzmodule.AppModule{}.ConsensusVersion(),
+		"staking":      staking.AppModule{}.ConsensusVersion(),
+		"mint":         mint.AppModule{}.ConsensusVersion(),
+		"distribution": distribution.AppModule{}.ConsensusVersion(),
+		"slashing":     slashing.AppModule{}.ConsensusVersion(),
+		"gov":          gov.AppModule{}.ConsensusVersion(),
+		"params":       params.AppModule{}.ConsensusVersion(),
+		"vesting":      vesting.AppModule{}.ConsensusVersion(),
+		"feegrant":     feegrantmodule.AppModule{}.ConsensusVersion(),
+		"evidence":     evidence.AppModule{}.ConsensusVersion(),
+		"crisis":       crisis.AppModule{}.ConsensusVersion(),
+		"genutil":      genutil.AppModule{}.ConsensusVersion(),
+		"capability":   capability.AppModule{}.ConsensusVersion(),
+		"blob":         blob.AppModule{}.ConsensusVersion(),
+		"qgb":          qgb.AppModule{}.ConsensusVersion(),
+		"ibc":          ibc.AppModule{}.ConsensusVersion(),
+		"transfer":     transfer.AppModule{}.ConsensusVersion(),
+	}
```

Comment:
The way we map app versions to module versions will need to be modified later on as it doesn't quite fit to how the SDK are used to doing migrations. I will follow up with this in a later PR but for now this makes the tests pass and somewhat signals how migrations will work

Reply:
I can dig it